### PR TITLE
all: refactor trie API

### PR DESCRIPTION
**Pull request by rjl493456442**
In this PR, all TryXXX(e.g. TryGet) APIs of trie are renamed to XXX(e.g. Get) with an error returned.

The original XXX(e.g. Get) APIs are renamed to MustXXX(e.g. MustGet) which will panic in case any error occurs.

### CODE REVIEW

**Comment by rjl493456442** [comment]
Note, this PR has a behavioral change, some APIs will panic if any error occurs instead of printing an error log.
I believe it will surface some corner case we have in the codebase. 

**But please don't merge this PR unless we know that all corner cases are addressed!**

File: trie/stacktrie.go [hunk 1]
```diff
@@ -202,8 +202,8 @@ const (
 	hashedNode
 )
 
-// TryUpdate inserts a (key, value) pair into the stack trie
-func (st *StackTrie) TryUpdate(key, value []byte) error {
+// Update inserts a (key, value) pair into the stack trie
```
Comment (MariusVanDerWijden) [inline trie/stacktrie.go:205]:
```suggestion
// Update inserts a (key, value) pair into the stack trie.
```

**Comment by holiman** [comment]
> In this PR, all TryXXX(e.g. TryGet) APIs of trie are renamed to XXX(e.g. Get) with an error returned.


Would it make sense to do that as a first PR? Since it's a pure rename, we can easily get that out of the way, to focus the reviewing on

> The original XXX(e.g. Get) APIs are renamed to MustXXX(e.g. MustGet) which will panic in case any error occurs.

**Comment by rjl493456442** [comment]
> > In this PR, all TryXXX(e.g. TryGet) APIs of trie are renamed to XXX(e.g. Get) with an error returned.
> 
> Would it make sense to do that as a first PR? Since it's a pure rename, we can easily get that out of the way, to focus the reviewing on
> 
> > The original XXX(e.g. Get) APIs are renamed to MustXXX(e.g. MustGet) which will panic in case any error occurs.

Fair enough. I will make the change.

**Review by Hektoreth — APPROVED** [review]
Ok

File: trie/trie.go [hunk 2]
```diff
@@ -231,41 +246,37 @@ func (t *Trie) tryGetNode(origNode node, path []byte, pos int) (item []byte, new
...
-//
-// The value bytes must not be modified by the caller while they are
-// stored in the trie.
-func (t *Trie) Update(key, value []byte) {
-	if err := t.TryUpdate(key, value); err != nil {
+// MustUpdate is a wrapper of Update and will omit any encountered error but
+// just print out an error message.
+func (t *Trie) MustUpdate(key, value []byte) {
+	err := t.Update(key, value)
+	if err != nil {
```
Comment (holiman) [inline trie/trie.go:261]:
```suggestion
	if err := t.Update(key, value); err != nil {
```

File: trie/trie.go [hunk 3]
```diff
@@ -363,16 +374,20 @@ func (t *Trie) insert(n node, prefix, key []byte, value node) (bool, node, error
...
 }
 
-// Delete removes any existing value for key from the trie.
-func (t *Trie) Delete(key []byte) {
-	if err := t.TryDelete(key); err != nil {
+// MustDelete is a wrapper of Delete and will omit any encountered error but
+// just print out an error message.
+func (t *Trie) MustDelete(key []byte) {
+	err := t.Delete(key)
+	if err != nil {
```
Comment (holiman) [inline trie/trie.go:381]:
```suggestion	
	if err := t.Delete(key); err != nil {
```

File: trie/secure_trie.go [hunk 4]
```diff
@@ -161,18 +160,20 @@ func (t *StateTrie) UpdateAccount(address common.Address, acc *types.StateAccoun
...
 }
 
-// Delete removes any existing value for key from the trie.
-func (t *StateTrie) Delete(key []byte) {
-	if err := t.DeleteStorage(common.Address{}, key); err != nil {
-		log.Error("Unhandled trie error in StateTrie.Delete", "err", err)
-	}
+// MustDelete removes any existing value for key from the trie.
+//
+// This function will panic if any error is encountered.
```
Comment (holiman) [inline trie/secure_trie.go:172]: Will it really? Won't it just log the error?

**Review by holiman — COMMENTED** [review]
I'm 99% certain this is good to go

File: trie/secure_trie.go (hunk 4 above)
Comment (rjl493456442) [reply trie/secure_trie.go:172]: I forgot to change the comment, will fix it.

**Review by holiman — APPROVED** [review]
LGTM

**Comment by holiman** [comment]
OBS

> Note, this PR has a behavioral change, some APIs will panic if any error occurs instead of printing an error log.
> I believe it will surface some corner case we have in the codebase. 
> 
> **But please don't merge this PR unless we know that all corner cases are addressed!**

This ^ is no longer true, this PR is a straight-up refactor without (intentional) breaking changes.

**Comment by donatocayurin** [comment]
Thank you ❤️🙏

**Comment by donatocayurin** [comment]
❤️🙏

**Pull request merged**